compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -3470,6 +3472,28 @@ private STNode parseFuncCall(STNode identifier) {
         return STNodeFactory.createFunctionCallExpressionNode(identifier, openParen, args, closeParen);
     }
 
+    /**
+     * Parse error constructor expression.
+     * <code>functional-constructor-expr := functionally-constructible-type-reference ( arg-list )
+     * functionally-constructible-type-reference := error | type-reference</code>
+     *
+     * @param kind Syntax kind
+     * @return Functional constructor expression
+     */
+    private STNode parseFunctionalConstructorExpr(SyntaxKind kind) {
```

Comment:
Do we need the syntaxKind as a parameter for this? This method is used only in one place

Reply:
At this moment we may have only one SyntaxKind for functional-constructor-expr. But with the implementation of other type references, new SyntaxKinds can be introduced.

Reply:
We can update the method when we have the requirement. Reason is, if anyone implementing the other type-reference decided not to use this method, then this change would become unnecessary.

Reply:
We should also look at the possibility of using the function-call node to represent error-constructor.

Reply:
Removed the syntax kind and it was handled same as FunctionCall.